### PR TITLE
[script.skipit] 1.0.1

### PR DESCRIPTION
**Pull request by shartrec**
### Description
Minor bugfix and Improved handling when seek beyond the start/end of file or when player is not playing.
### Checklist:

- [X ] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [ X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [ X] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: script.skipit/resources/lib/skipit.py [hunk 1]
```diff
@@ -67,23 +67,54 @@ def onAction(self, action):
...
     reset_timer()
     halve()
     set_label(skip_secs, True)
-    command = build_command(skip_secs - slippage, True)
-    xbmc.executebuiltin(command)
+    requested_skip = skip_secs - slippage
+    actual_skip = do_skip(requested_skip)
+    if actual_skip != requested_skip:
+        skip_secs = abs(actual_skip)
+        command = 'Notification({},{})'
```
Comment (basrieter) [inline script.skipit/resources/lib/skipit.py:105]: Why not use  ` xbmcgui.Dialog().notification(heading, message[, icon, time, sound])`: https://codedocs.xyz/MartijnKaijser/xbmc/group__python___dialog.html#ga27157f98dddb21b44cc6456137977aa2

File: script.skipit/resources/lib/skipit.py [hunk 2]
```diff
@@ -92,8 +123,15 @@ def skip_backward():
...
     set_label(skip_secs, False)
     # Add a couple of extra seconds as the video is still playing forwards
-    command = build_command(skip_secs + slippage, False)
-    xbmc.executebuiltin(command)
+    requested_skip = - skip_secs + slippage
+    actual_skip = do_skip(requested_skip);
+    if actual_skip != requested_skip:
+        skip_secs = abs(actual_skip)
+        text = ADDON.getLocalizedString(32130)
+        command = 'Notification({},{})'
```
Comment (basrieter) [inline script.skipit/resources/lib/skipit.py:131]: Why not use `xbmcgui.Dialog().notification(heading, message[, icon, time, sound])`: https://codedocs.xyz/MartijnKaijser/xbmc/group__python___dialog.html#ga27157f98dddb21b44cc6456137977aa2

**Review by basrieter — CHANGES_REQUESTED** [review]
Please check my comments on `xbmcgui.Dialog().notification(heading, message[, icon, time, sound])`

File: script.skipit/resources/lib/skipit.py (hunk 1 above)
Comment (shartrec) [reply script.skipit/resources/lib/skipit.py:105]: I would prefer that. Previously I couldn't fathom the api for that one. Thanks

File: script.skipit/resources/lib/skipit.py (hunk 2 above)
Comment (shartrec) [reply script.skipit/resources/lib/skipit.py:131]: I would prefer that. Previously I couldn't fathom the api for that one. Thanks

**Comment by shartrec** [comment]
Addon-Checker failed with a networking error. I'm not sure how to proceed. Do I need to do anything, or should this correct itself?

File: script.skipit/resources/lib/skipit.py [hunk 3]
```diff
@@ -67,35 +68,69 @@ def onAction(self, action):
           reset()
           skip_backward()
 
+
+def do_skip(skip_secs):
+    # We need to track the actual skip, in case we run past the end
+    totalTime = xbmc.Player().getTotalTime()
```
Comment (enen92) [inline script.skipit/resources/lib/skipit.py:74]: Maybe consider reusing the same player instance in the method scope? You're initializing three instances for no reason

Comment (shartrec) [reply script.skipit/resources/lib/skipit.py]: Done. :)

**Review by enen92 — APPROVED** [review]

**Pull request merged**